Makefile.gen

```diff
@@ -24,10 +24,32 @@ symlink-gopath:
 	@[[ $(shell pwd) == $(shell realpath $(CANONICAL_GOPATH)) ]] \
 		|| (echo "expected $(shell pwd) to equal $(shell realpath $(CANONICAL_GOPATH))" ; exit 1)
 
+CLIENTGEN_TOOLS := client-gen deepcopy-gen informer-gen lister-gen conversion-gen
+# This should match the k8s.io/code-generator version in go.mod.
+CLIENTGEN_VERSION := "v0.22.2"
+
+.PHONY: $(CLIENTGEN_TOOLS)
+$(CLIENTGEN_TOOLS):
+	GOPATH="$(GO_DIR)" go install "k8s.io/code-generator/cmd/$(@)@$(CLIENTGEN_VERSION)"
+
+.PHONY: install-clientgen-tools
+# install all clientgen tools (user-friendly target alias)
+install-clientgen-tools: $(CLIENTGEN_TOOLS)
+
+CLEAN_CLIENTGEN_TOOLS := $(patsubst %,clean-%,$(CLIENTGEN_TOOLS))
+
+.PHONY: $(CLEAN_CLIENTGEN_TOOLS)
+$(CLEAN_CLIENTGEN_TOOLS):
```

Comment:
hmm. This clean isn't human friendly any more. I don't think we need a non-human friendly one, because we only ever run the top level clean to nuke the .output dir. 

i don't know how much it matters. it's just inconsistent, since this is the first one doing a whole list.

Reply:
The human friendly one is below